lib/logging/sentry-transport.js

```diff
@@ -0,0 +1,51 @@
+const TransportStream = require('winston-transport');
+const Sentry = require('@sentry/node');
+
+const config = require('../config');
+
+
+module.exports = class SentryTransport extends TransportStream {
+
+  constructor(dsn, options = {}) {
+    options.level = options.level || 'error';
+    super(options);
+
+    this.name = options.name || 'SentryTransport';
+    this.Sentry = Sentry;
+    this.Sentry.init({
+      dsn: dsn,
+      debug: true,
+      environment: config.environment,
+      release: process.env.npm_package_version
+    });
+    this.Sentry.configureScope(scope => {
+      scope.setTag('nodeVersion', process.version);
+    })
+  }
+
+  log({message, metadata, ...extra}, callback) {
+    setImmediate(() => this.emit('logged', {message, metadata, ...extra}));
+    let error = extra['error'] || metadata['error'] || metadata['originalError'];
+
+    if (error && error instanceof Error) {
+      delete extra['error'];
+      delete metadata['error'];
+      delete metadata['originalError'];
+    } else {
+      error = new Error(message);
+      error.stack = metadata['stack'];
+    }
+    delete metadata['stack'];
+
+    this.Sentry.configureScope(scope => {
+      Object.keys(extra).forEach((key) => {
+        scope.setExtra(key, extra[key]);
+      });
+      Object.keys(metadata).forEach((key) => {
+        scope.setExtra(key, metadata[key]);
+      });
+      this.Sentry.captureException(error);
+    });
+    return callback();
```

Comment:
above you've called this with `new SentryTransport(config.sentryDSN, {level: 'error'})` which means `callback` is `undefined`. would that throw an error here? maybe we should be more defensive here?

Reply:
The callback is passed to `log` function by `winston`.  It won't be `undefined`.